lib/src/facts/linux/filesystem_resolver.cc

```diff
@@ -75,10 +75,9 @@ namespace facter { namespace facts { namespace linux {
         char buffer[4096];
         while (mntent *ptr = getmntent_r(file, &entry, buffer, sizeof(buffer))) {
             string device = ptr->mnt_fsname;
-            string mtype = ptr->mnt_type;
 
-            // Skip over anything that doesn't map to a device and is not tmpfs
-            if (!boost::starts_with(device, "/dev/") && mtype != "tmpfs") {
+            // Skip over any mount under /proc or /sys
+            if (boost::starts_with(device, "/proc") || boost::starts_with(device, "/sys")) {
```

Comment:
This won't work also because you are trying to  skip if the device matches the /sys or /proc directory. You can see what fields getmntent_r returns here: http://ftp.linuxfoundation.org/pub/lsb/lsb-openjdk/openjdk7-lsb-overrides/include/mntent.h 

i thinks in order that this fix to work the statement should be : 
`if (!boost::starts_with(device, "/dev/") && mtype != "tmpfs" && mtype.find("nfs") == string::npos) {
                continue;
  }`

Also please rebase on 3.11.x as you are still on puppetlabs:master

Reply:
This change is to have facter return non-tmpfs mounts not under /dev. Adding an specific exception for nfs type mounts won't address all the problems.